# <font color= #79b0e9> **Examen Parcial 3** </font> 

Sofía Maldonado García

Viviana Toledo De la Fuente

Iván Morales

"Y aquí iría le 4to integrante, si tuviéramos une"

In [1]:
# Librerías Generales
import numpy as np
import pandas as pd

# Gráficas
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go
import seaborn as sns

# Modelado
from sklearn.linear_model import LinearRegression, Ridge
from sklearn.svm import SVR

# Métricas
from sklearn.metrics import r2_score, mean_squared_error

# Tratamiento de Datos
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import Pipeline
from sklearn.model_selection import GridSearchCV
from scipy.stats import boxcox
from sklearn.model_selection import train_test_split

---

# <font color= #79b0e9> **Análisis Exploratorio de Datos** </font> 

In [10]:
df = pd.read_csv('data/train_data.csv')
print(f' Tamaño del dataset: {df.shape}')
df.head()

 Tamaño del dataset: (119, 10)


Age     Sex  TCh  HCh  LCh   Tr  BPS          Sm Diabetes  Pred
0   50    Male  272   68  188   82  124  Non-smoker       No  High
1   29  Female  275   68  183   84  129  Non-smoker       No  High
2   50    Male  268   55  189  111  119  Non-smoker       No  High
3   78  Female  194   50  120   94  113  Non-smoker       No  High
4   63    Male  229   35  182  214  160      Smoker      Yes  High

Contamos con un dataset de 119 observaciones, 9 features y 1 target.

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 119 entries, 0 to 118
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Age       119 non-null    int64 
 1   Sex       119 non-null    object
 2   TCh       119 non-null    int64 
 3   HCh       119 non-null    int64 
 4   LCh       119 non-null    int64 
 5   Tr        119 non-null    int64 
 6   BPS       119 non-null    int64 
 7   Sm        119 non-null    object
 8   Diabetes  119 non-null    object
 9   Pred      119 non-null    object
dtypes: int64(6), object(4)
memory usage: 9.4+ KB


Tenemos 6 features numéricas y 3 features categóricas (sex, smoking y diabetes). Tendremos que codificar estas variables para tomarlas en cuenta en nuestro modelo.

# <font color= #79b0e9> **2. Support Vector Classifier** </font> 